core/src/main/scala/org/apache/spark/rdd/BinaryFileRDD.scala

```diff
@@ -35,8 +36,12 @@ private[spark] class BinaryFileRDD[T](
   extends NewHadoopRDD[String, T](sc, inputFormatClass, keyClass, valueClass, conf) {
 
   override def getPartitions: Array[Partition] = {
-    val inputFormat = inputFormatClass.newInstance
     val conf = getConf
+    // setMinPartitions below will call FileInputFormat.listStatus(), which can be quite slow when
+    // traversing a large number of directories and files. Parallelize it.
+    conf.setIfUnset(FileInputFormat.LIST_STATUS_NUM_THREADS,
+      Runtime.getRuntime.availableProcessors().toString)
```

Comment:
@cloud-fan @kiszk this will happen on the driver, where it goes to assess the size of the data in order to compute the desired number of partitions. I don't know what the right value is here, to be honest. It won't actually use a ton of CPU because these threads will mostly be waiting for the external FS processes.

I didn't think it's worth yet another config, and didn't know what else to base it on. The number of driver cores maybe?

What is `CPU_CORES_PER_EXECUTOR` by the way, I don't see that in the code? I can check for `spark.driver.cores` in the Spark config here.